src/main/java/bio/terra/service/resourcemanagement/google/GoogleProjectService.java

```diff
@@ -401,6 +418,102 @@ private CloudResourceManager cloudResourceManager() throws IOException, GeneralS
+        JsonFactory jsonFactory = JacksonFactory.getDefaultInstance();
+
+        GoogleCredential credential = GoogleCredential.getApplicationDefault();
+        if (credential.createScopedRequired()) {
+            credential =
+                credential.createScoped(
+                    Collections.singletonList("https://www.googleapis.com/auth/cloud-platform"));
+        }
+
+        return new Appengine.Builder(httpTransport, jsonFactory, credential)
+            .setApplicationName(resourceConfiguration.getApplicationName())
+            .build();
+    }
+
+    /**
+     * Enable Firestore in native mode in an existing project
+     * @param appengine appengine client
+     * @param googleProjectId name of the google project to create the Firestore DB in
+     * @param locationId location of the Firestore DB
+     * @param timeout how long to wait for the creation operation
+     */
+    private static void enableFirestore(final Appengine appengine,
+                                        final String googleProjectId,
+                                        final String locationId,
+                                        final long timeout) throws IOException, InterruptedException {
+        logger.info("Enabling Firestore in project {} in location {}", googleProjectId, locationId);
+        // Create a request object
+        Appengine.Apps.Create createRequest = appengine.apps().create(new Application()
+            .setId(googleProjectId)
+            .setLocationId(locationId)
+            .setDatabaseType(FIRESTORE_DB_TYPE)
+        );
+
+        // Make sure that Firestore is created in the correct project
+        createRequest.getRequestHeaders().set("X-Goog-User-Project", googleProjectId);
+
+        // Execute the request
+        com.google.api.services.appengine.v1.model.Operation operation = createRequest.execute();
+
+        // Wait for the Firestore creation to finish
+        blockUntilAppengineOperationComplete(appengine, operation, googleProjectId, timeout);
+        logger.info("Firestore was enabled successfully");
+    }
+
+    /**
+     * Poll the app engine api until an operation completes. It is possible to hit quota issues
+     * here, so the timeout is set to 10 seconds.
+     *
+     * @param appengine       service instance
+     * @param operation       has an id for us to use in the check
+     * @param appId           id of the app that the operation is related to
+     * @param timeoutSeconds  how many seconds before we give up
+     * @return a completed operation
+     */
+    private static com.google.api.services.appengine.v1.model.Operation blockUntilAppengineOperationComplete(
+        Appengine appengine,
+        com.google.api.services.appengine.v1.model.Operation operation,
+        String appId,
+        long timeoutSeconds) throws IOException, InterruptedException {
+        long start = System.currentTimeMillis();
+        final long pollInterval = 10 * 1000; // 10 seconds
```

Comment:
Use `TimeUnit` to convert seconds to millis
```suggestion
        final long pollInterval = TimeUnit.SECONDS.toMillis(10);
```

Reply:
Yeah, fair point.  I'll go an update all of the places in here that we do this